backend/routes/run.js

```diff
@@ -0,0 +1,194 @@
+
+  const newPlayer = await prisma.player.create({
+    data: {
+      AST: player.BLK,
+      DRB: player.DRB,
+      ORB: player.ORB,
+      PF: player.PF,
+      PTS: player.PTS,
+      STL: player.STL,
+      TOV: player.TOV,
+      TRB: player.TRB,
+      BLK: player.BLK,
+      age: player.age,
+      effect_fg_percent: effect_fg_percent,
+      field_attempts: player.field_attempts,
+      field_goals: player.field_goals,
+      field_percent: field_percent,
+      ft: player.ft,
+      ft_percent: ft_percent,
+      fta: player.fta,
+      games: player.games,
+      games_started: player.games_started,
+      id: player.id,
+      minutes_played: player.minutes_played,
+      player_name: player.player_name,
+      team: player.team,
+      three_attempts: player.three_attempts,
+      three_fg: player.three_fg,
+      three_percent: three_percent,
+      two_attempts: player.two_attempts,
+      two_fg: player.two_fg,
+      two_percent: two_percent,
+    },
+  });
+  return newPlayer;
+}
+
+async function changePlayerData(player, oldPlayer) {
+  if (!isEqual(player, oldPlayer)) {
+    const field_percent =
+      player.field_attempts === 0
+        ? 0
+        : player.field_goals / player.field_attempts;
+    const ft_percent = player.fta === 0 ? 0 : player.ft / player.fta;
+    const three_percent =
+      player.three_attempts === 0 ? 0 : player.three_fg / player.three_attempts;
+    const two_percent =
+      player.two_attempts === 0 ? 0 : player.two_fg / player.two_attempts;
+    const effect_fg_percent = !player.effect_fg_percent
+      ? "0"
+      : player.effect_fg_percent;
+
+    player = await prisma.player.update({
+      where: {
+        id: oldPlayer.id,
+      },
+      data: {
+        AST: player.BLK,
+        DRB: player.DRB,
+        ORB: player.ORB,
+        PF: player.PF,
+        PTS: player.PTS,
+        STL: player.STL,
+        TOV: player.TOV,
+        TRB: player.TRB,
+        BLK: player.BLK,
+        age: player.age,
+        effect_fg_percent: effect_fg_percent,
+        field_attempts: player.field_attempts,
+        field_goals: player.field_goals,
+        field_percent: field_percent,
+        ft: player.ft,
+        ft_percent: ft_percent,
+        fta: player.fta,
+        games: player.games,
+        games_started: player.games_started,
+        id: player.id,
+        minutes_played: player.minutes_played,
+        player_name: player.player_name,
+        team: player.team,
+        three_attempts: player.three_attempts,
+        three_fg: player.three_fg,
+        three_percent: three_percent,
+        two_attempts: player.two_attempts,
+        two_fg: player.two_fg,
+        two_percent: two_percent,
+      },
+    });
+    return player;
+  }
+  return oldPlayer;
+}
+
+function isEqual(player, oldPlayer) {
+  return (
+    player.player_name === oldPlayer.player_name &&
+    player.age === oldPlayer.age &&
+    player.minutes_played === oldPlayer.minutes_played
```

Comment:
What's the need for this function when we already query the `oldPlayer` by id? I can see this breaking when  someone's age changes or they play a new game right

Reply:
That function is meant to check if we actually need to update an existing player's information. Before this point, we only check to see if the player is in the database at all in updatePlayer (oldPlayer === null). If they are, we reach this point, and check if there's anything that actually needs to be updated (if the old player and the new player differ by either name, age, or games played). So it's really just a quick condition that allows us to skip some data manipulation for players whose information doesn't need to be touched at all.